test/parallel/test-async-local-storage-snapshot.js

```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+require('../common');
+const { strictEqual } = require('assert');
+const { AsyncLocalStorage } = require('async_hooks');
+
+const asyncLocalStorage = new AsyncLocalStorage();
+const runInAsyncScope = asyncLocalStorage.run(123, () => AsyncLocalStorage.snapshot());
+const result = asyncLocalStorage.run(321, () => runInAsyncScope(() => asyncLocalStorage.getStore()));
+strictEqual(result, 123);
```

Comment:
Not strictly necessary since the `strictEqual(result, 123)` would fail if the callbacks aren't called, but for consistency across other tests, it's a good idea to use `common.mustCall()` here

```suggestion
const common = require('../common');
const { strictEqual } = require('assert');
const { AsyncLocalStorage } = require('async_hooks');

const asyncLocalStorage = new AsyncLocalStorage();
const runInAsyncScope = asyncLocalStorage.run(123,
  common.mustCall(() => AsyncLocalStorage.snapshot()));
const result = asyncLocalStorage.run(321,
  common.mustCall(() => runInAsyncScope(() => asyncLocalStorage.getStore())));
strictEqual(result, 123);
```